scrapy/extensions/feedexport.py

```diff
@@ -242,7 +242,7 @@ def open_spider(self, spider):
     def close_spider(self, spider):
         slot = self.slot
         if not slot.itemcount and not self.store_empty:
-            return
+            return defer.maybeDeferred(slot.storage.store, slot.file)
```

Comment:
Could you please explain how this fix works?

Reply:
In `FeedExporter.open_spider` we are opening the output file, regardless of the value of `self.store_empty`. `slot.storage.store` is responsible for closing that file. See `scrapy.extensions.feedexport.FileFeedStorage.store`.
The code here, however, was skipping that step when there was no item to store and storing an empty file was disabled.

Reply:
Aha, this makes sense. Could you please add a small comment here though? Now I see that this is a copy-paste of the code below (`d = defer.maybeDeferred(slot.storage.store, slot.file)`). But it is not obvious, from these few lines themselves, that slot.storage.store closes a file.